### PR TITLE
Check release tag

### PR DESCRIPTION
**Pull request by 8W9aG**
Check that the release tag is a version tag (v[0-9].[0-9]+.[0-9]+) and that the workflow is triggered by a new tag push.

### CODE REVIEW

**Comment by mattt** [comment]
@8W9aG I think we need to update this line:

https://github.com/replicate/cog/blob/4efdcf386ec3f83c073d43ff9fcf316172b33773/.github/workflows/release.yaml#L9

If you look at past runs, the Release workflow only ran for merge commits to `main`. The tags triggered CI as expected, but not Release.

**Comment by 8W9aG** [comment]
> @8W9aG I think we need to update this line:
> 
> https://github.com/replicate/cog/blob/4efdcf386ec3f83c073d43ff9fcf316172b33773/.github/workflows/release.yaml#L9
> 
> If you look at past runs, the Release workflow only ran for merge commits to `main`. The tags triggered CI as expected, but not Release.

Done, let me know how its looking!

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -23,8 +22,7 @@ jobs:
       - name: Check if commit has a version tag
         id: check-tag
         run: |
-          LATEST_TAG=$(git tag | sort --version-sort | tail -n1)
-          if [[ ${{ github.ref_name == $LATEST_TAG }} -a $LATEST_TAG =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+          if [[ ${{ github.event.workflow_run.head_branch }} =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
```
Comment (mattt) [inline .github/workflows/release.yaml]: `github.event.workflow_run.head_branch` 🤨 

Weird, seems to be right.

**Review by mattt — APPROVED** [review]
Thanks for chasing this down, @8W9aG 🙏

**Pull request merged**